gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/PolicyWriter.java

```diff
@@ -49,22 +51,98 @@ public void write(Bundle bundle, File rootFolder) {
         });
 
         //create policies
+        Map<String, PolicyMetadata> policyMetadataMap = new HashMap<>();
         Map<String, Service> services = bundle.getEntities(Service.class);
-        services.values().parallelStream().forEach(serviceEntity -> writePolicy(bundle, policyFolder, serviceEntity.getParentFolder().getId(), serviceEntity.getName(), serviceEntity.getPolicyXML()));
+        services.values().parallelStream().forEach(serviceEntity -> {
+            final String id = serviceEntity.getId();
+            final String name = serviceEntity.getName();
+            final Folder folder = bundle.getFolderTree().getFolderById(serviceEntity.getParentFolder().getId());
+            final Path policyPath = bundle.getFolderTree().getPath(folder);
+            writePolicy(policyFolder, policyPath, name, serviceEntity.getPolicyXML());
+            final PolicyMetadata policyMetadata = createPolicyMetadata(id, rawBundle, null);
+            final String path = PathUtils.unixPathEndingWithSeparator(policyPath);
+            final String key = path.length() > 1 ? path + name : name;
+            policyMetadataMap.put(key, policyMetadata);
+        });
 
         Stream.of(
                 bundle.getEntities(Policy.class).values().stream(),
                 bundle.getEntities(GlobalPolicy.class).values().stream().map(Policy.class::cast).collect(toList()).stream(),
                 bundle.getEntities(AuditPolicy.class).values().stream().map(Policy.class::cast).collect(toList()).stream()
         ).flatMap(s -> s)
-                .forEach(policyEntity -> writePolicy(bundle, policyFolder, policyEntity.getParentFolder().getId(), policyEntity.getName(), policyEntity.getPolicyDocument()));
+                .forEach(policyEntity -> {
+                    final String id = policyEntity.getId();
+                    final String name = policyEntity.getName();
+                    final Folder folder = bundle.getFolderTree().getFolderById(policyEntity.getParentFolder().getId());
+                    final Path policyPath = bundle.getFolderTree().getPath(folder);
+                    writePolicy(policyFolder, policyPath, name, policyEntity.getPolicyDocument());
+                    final PolicyMetadata policyMetadata = createPolicyMetadata(id, rawBundle, policyEntity);
+                    final String path = PathUtils.unixPathEndingWithSeparator(policyPath);
+                    final String key = path.length() > 1 ? path + name : name;
+                    policyMetadataMap.put(key, policyMetadata);
```

Comment:
Can you merge this code with the above one and avoid code duplication?

Reply:
done